sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/DetectNewPartitionsAction.java

```diff
@@ -146,15 +146,22 @@ private ProcessContinuation schedulePartitions(
       OutputReceiver<PartitionMetadata> receiver,
       Timestamp minWatermark,
       TreeMap<Timestamp, List<PartitionMetadata>> batches) {
+    List<PartitionMetadata> batchPartitionsDifferentCreatedAt = new ArrayList<>();
+    int numTimestampsHandledSofar = 0;
     for (Map.Entry<Timestamp, List<PartitionMetadata>> batch : batches.entrySet()) {
+      numTimestampsHandledSofar++;
       final Timestamp batchCreatedAt = batch.getKey();
-      final List<PartitionMetadata> batchPartitions = batch.getValue();
-
-      final Timestamp scheduledAt = updateBatchToScheduled(batchPartitions);
-      if (!tracker.tryClaim(batchCreatedAt)) {
-        return ProcessContinuation.stop();
+      final List<PartitionMetadata> batchPartitionsSameCreatedAt = batch.getValue();
+      batchPartitionsDifferentCreatedAt.addAll(batchPartitionsSameCreatedAt);
+      if (batchPartitionsDifferentCreatedAt.size() >= 200
```

Comment:
Just to verify my understanding: we are trying to increase the batch size in order to reduce the number of database operations we need to do, is that correct?

It LGTM.